src/module-elasticsuite-virtual-category/Model/Rule.php

```diff
@@ -181,9 +181,9 @@ public function getSearchQueriesByChildren(CategoryInterface $rootCategory): arr
                 ->addIsActiveFilter()
                 ->addIdFilter($childrenIds)
                 ->addAttributeToSelect(['virtual_category_root', 'is_virtual_category', 'virtual_rule']);
-
+            $excludedCategories = [];
             foreach ($categoryCollection as $category) {
-                $childQuery = $this->getCategorySearchQuery($category);
+                $childQuery = $this->getCategorySearchQuery($category, $excludedCategories);
```

Comment:
Initializing/resetting the $excludedCategories should be done inside the loop.
=> This method is called to build the category facet/aggregation : each \Smile\ElasticsuiteVirtualCategory\Model\Rule::getCategorySearchQuery should therefore be independent, otherwise children virtual categories sharing the same virtual root category could simply "disappear" from the "categories" aggregation.

Reply:
Unfortunately moving the definition of the `$excludedCategories` variable inside the loop doesn't resolve the issue.
 - https://filebin.ca/58onD2YI91Rh/test_2020-01-14_inside-loop-incomplete.txt - incomplete log with `$excludedCategories` inside the loop. I reloaded the php-fpm after got the 504 Gateway Time-out error from nginx
 - https://filebin.ca/58on9IGIW7Kz/test_2020-01-14_before-loop.txt - full log with `$excludedCategories` outside loop